### DISCUSSION TITLE
Gathering non numeric values

### DISCUSSION THREAD

**Discussion by Arij-Aladel** [horovod/horovod#3485] [Q&A]
Hello! we are using hvd.allreduce to gather tensor values from different gpus. What if I have evaluation function which in result produces an answer dictionary for example {"0b014789":"yes", "0b458796":"in the school"} . Naturally we will have multiple dictionaries, is there an example how to gather these values from multiple gpus? I hope I expressed my idea clearly

**Comment by maxhgerlach** [comment]
Does `allgather_object` provide what you are looking for?

**Comment by Arij-Aladel** [comment]
I am using torch, yesterday I read on the internet that gathering dictionaries accross multiple gpus is not preferable and it is better to write them to csv file. So I wrote a script for testing on just one gpu after all. thanks for you reply.
